### PR TITLE
Firebase and more

### PR DESCRIPTION
**Pull request by austBrink**
# message popup, Firebase, login, logout, abstracted components, 
- set up firebase utils 
- set up a login page 
- create a OurComponents file that exports little component modules we can make to standardize things like input and label (TextField) 
- use in login to employ firebase auth 
- created a message component to flash messages for user on screen 
- header with left right and middle content (flex)
- logout using firebase and setting context user to {}

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -1,3 +1,9 @@
+html {
```
Comment (austBrink) [inline src/App.css:1]: We can maybe put our variable in separate css files. this works for me for now.

File: src/components/login/Login.jsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    const handleSubmit = (e) => {
+        e.preventDefault();
+        setFormErrors({...validate(formValues)});
+        setIsSubmitting(true);
+    }
+
+    const validate = () => {
+        const errors = {}
+        return {};
+        // for(key in formValues) {
```
Comment (austBrink) [inline src/components/login/Login.jsx:53]: needs to be 
```
for (const [ key, value ] of Object.entries(values)) {
...
}
```

File: src/App.js [hunk 3]
```diff
@@ -1,13 +1,47 @@
...
+// import Signup from './components/signup/Signup';
+import Message from './components/message/Message';
+import UserContext from './context/UserContext';
 
 function App() {
+
+  const [ user, setUser ] = useState({});
+  const [ message, setMessage ] = useState(null); 
+
+  const popMessage = (type, message) => {
```
Comment (austBrink) [inline src/App.js:29]: not totally done with this. Need an option to close it in case it annoys the user.

**Review by austBrink — COMMENTED** [review]
A WIP but still ready to push to main

**Review by austBrink — COMMENTED** [review]
Guess I wasn't done. 💣

File: src/App.css (hunk 1 above)
Comment (TeoTodorov16) [reply src/App.css:1]: Yeah that will be a good idea since it will be organised

**Review by TeoTodorov16 — COMMENTED** [review]
!

**Review by TeoTodorov16 — COMMENTED** [review]
!

**Pull request merged**